### PR TITLE
On demand report containing sam.gov data

### PR DESCRIPTION
**Pull request by jessieay**
Closes https://github.com/18F/micropurchase/issues/495

screen shot 2016-05-16 at 11 25 17 am

When DUNS is not valid, shows a flash message saying that the DUNS is not valid

### CODE REVIEW

File: app/models/winning_bid.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class WinningBid
```
Comment (harrisj) [inline app/models/winning_bid.rb]: I'm not sure I understand why this needs to be a separate class since it's delegating for all of its functionality... Also, should it be in models since it isn't AR-backed?

File: app/models/winning_bidder_export.rb [hunk 2]
```diff
@@ -0,0 +1,118 @@
+require 'csv'
+
+class WinningBidderExport
```
Comment (harrisj) [inline app/models/winning_bidder_export.rb]: Nice class

File: config/application.rb [hunk 3]
```diff
@@ -9,35 +9,13 @@
 require "action_mailer/railtie"
 require "action_view/railtie"
 require "sprockets/railtie"
```
Comment (harrisj) [inline config/application.rb]: Thank you, but we should probably do this kind of cleanup in separate PRs?

File: features/step_definitions/auction_steps.rb [hunk 4]
```diff
@@ -35,11 +35,16 @@
 end
 
 When(/^I visit the auction page$/) do
-  visit "/auctions/#{@auction.id}"
+  visit auction_path(@auction)
```
Comment (harrisj) [inline features/step_definitions/auction_steps.rb]: Thank you!

File: spec/rails_helper.rb [hunk 5]
```diff
@@ -1,4 +1,4 @@
-ENV['RAILS_ENV'] = 'test'
+ENV['RAILS_ENV'] ||= 'test'
```
Comment (harrisj) [inline spec/rails_helper.rb]: Now this is a ||= again. Did we ever figure out why it was changed before? Just curious...

File: spec/rails_helper.rb [hunk 6]
```diff
@@ -31,6 +31,7 @@
 
   config.before do
     mock_github
+    WebMock.stub_request(:any, /api.data.gov/).to_rack(FakeSamApi)
```
Comment (harrisj) [inline spec/rails_helper.rb]: Ooooh, I didn't know we could do this!

File: spec/rails_helper.rb (hunk 5 above)
Comment (harrisj) [inline spec/rails_helper.rb]: Is this even needed considering this is also in the spec_helper.rb?

**Comment by harrisj** [comment]
I have a few questions, but nothing that should hold up a merge...

File: config/application.rb (hunk 3 above)
Comment (jessieay) [inline config/application.rb]: yes agreed, got overzealous!

File: app/models/winning_bid.rb (hunk 1 above)
Comment (jessieay) [inline app/models/winning_bid.rb]: we seem to have quite a few non-AR models. I am fine with putting anything model-y in `app/models`

I made it a separate class bc I realized I was instantiating the rules factory a bunch and then finding winning bidder based on that

File: spec/rails_helper.rb (hunk 5 above)
Comment (jessieay) [inline spec/rails_helper.rb]: oh d'oh I changed this when debugging a weird issue -- my rails console thinks it is in test env by default. can't figure out why!

File: spec/rails_helper.rb (hunk 6 above)
Comment (jessieay) [inline spec/rails_helper.rb]: yes! @adelevie and I worked on this approach when I was in DC and now we have a nice setup for stubbing out the SAM API
